### DISCUSSION TITLE
Refetch after mutation, fetchMore for pagination!

### DISCUSSION THREAD

**Discussion by siamahnaf** [urql-graphql/urql#3009] [Q&A] [answered]
Hello, Today I first time try urql. Before I tried to use apollo client. I stop using Apollo client because their server side issue on nextjs.

But when I use urql, it solve all serverside data fetching issues in nextjs.

But I am not finding any documentation about refetch after mutation, for updating list. Here I need also fetchMore option to add pagination. Is urql support those function `refetch after mutation` and `fetchMore`. Please five proper example of use case.

Thanks urql team.

**Comment by siamahnaf** [comment]
@kitten, can you place answer please?

**Comment by JoviDeCroock** [comment]
Why five proper examples? Honestly where did you try to find it in our documentation? As a maintainer I find it kind off disrepectful to make thise demands...

Especially because both caching ones explain it, doc-cache is typename based https://formidable.com/open-source/urql/docs/basics/document-caching/ and normalized caching is writing an update https://formidable.com/open-source/urql/docs/graphcache/cache-updates/

There are also numerous discussion threads and even examples in this repo for fetchMore and the two things you are asking for... in the examples folder

**Comment by kitten** [comment, marked as answer]
First of all, given the context, I assume you mean infinite pagination?

`fetchMore` and the likes is not how we implement pagination or recommend client-side pagination to be implemented. The mindset here being that it's a UI concern.
We have docs on how to implement pagination in various ways (ranging from "quick and dirty" solutions to what's actually recommended:
- https://formidable.com/open-source/urql/docs/basics/ui-patterns/#infinite-scrolling
- https://formidable.com/open-source/urql/docs/graphcache/local-resolvers/#paginationhttps://formidable.com/open-source/urql/docs/graphcache/local-resolvers/#pagination

We'd always recommend the former so that it's UI-driven, mostly.
For refetching in general, we have docs here:
- https://formidable.com/open-source/urql/docs/basics/react-preact/#reexecuting-queries
If you need invalidation of lists instead, that's documented here:
- https://formidable.com/open-source/urql/docs/graphcache/cache-updates/#invalidating-entities

If you need examples to get started with, have a look here:
- https://github.com/urql-graphql/urql/tree/main/examples/with-pagination
- or for the quick Graphcache solution: https://github.com/urql-graphql/urql/tree/main/examples/with-graphcache-pagination

In general, for invalidation with the normalised cache, you're looking at Graphcache's `cache.invalidate` after mutations. if you just need to refetch pages, reexecuting is good. Or, if you just need it to refetch periodically you can use this:
- https://github.com/urql-graphql/urql/tree/main/exchanges/request-policy

There's a lot of prior issues and discussions available on this. Please use the resources available (above), search for other issues, discussions, or docs as appropriate.

Now, lastly, please adhere to the Code of Conduct and guidelines for this repo. I can accept an RFC that's accidentally opened for a question; no problem. But then `@`-ing me right away is just rude. I'm not sitting at a support desk as a personal search engine. Please try to use the resources we have before just going ahead like you have. ❤️

**Reply by siamahnaf** [reply]
Sorry boss, I didn't mean that. I am actually seeking one's help. Thank you very much.
